src/classes/TERM_CannotDelete_TEST.cls

```diff
@@ -0,0 +1,102 @@
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Terms
+* @group-content ../../ApexDocContent/Terms.htm
+* @description Unit tests for TERM_CannotDelete_TDTM. These tests
+* make sure Terms cannot be deleted when they have Term Grade Associations.
+*/
+
+
+@isTest
+private class TERM_CannotDelete_TEST {
+
+    @testSetup
+    static void dataSetup() {
+        
+        // Insert Account
+        List<Account> academicAccount = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getAcademicAccRecTypeID());
+        insert academicAccount;
+        
+        // Insert Term
+        Term__c term = UTIL_UnitTestData_TEST.getTerm(academicAccount[0].Id, 'Spring 20');
+        insert term;
+    }
+    
+    // Test deletion of Terms with Term Grades
+
+    @isTest 
+    static void testDeleteTermsWithTermGrades() {
+        
+        // Insert Course Offering
+        Course_Offering__c courseOffering = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+        
+        // Insert Contact
+        Contact student = UTIL_UnitTestData_TEST.getContact();
+        student.LastName = 'Term Grade';
+        student.Email = 'termGrade@salesforce.org';
+        student.Phone = '12341234';   
+        insert student;
+        
+        // Insert Course Connection
+        Course_Enrollment__c courseConnection = UTIL_UnitTestData_TEST.getCourseConnection(student.Id, courseOffering.Id);
+        insert courseConnection;
+        
+        Term__c term = [SELECT Id FROM Term__c LIMIT 1];
+        
+        // Insert Term Grade
+        Term_Grade__c termGrade = new Term_Grade__c();
```

Comment:
Can we add a new getTermGrade() method to UTIL_UnitTestData_TEST that accepts these two parameters?

(Actually, it looks like we're adding a getTermGrade() method in one of the other child PRs, but that one has different parameters so I think having both methods would be fine.)